Fix aria-haspopup usage in the admin menu.

afercia opened the pull request:
<!--
Hi there! Thanks for contributing to WordPress!

Pull Requests in this GitHub repository **must** be linked to a ticket in the WordPress Core Trac instance (https://core.trac.wordpress.org), and are only used for code review. **No pull requests will be merged on GitHub.**

See the WordPress Handbook page on using PRs for Code Review more information: https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/

If this is your first time contributing, you may also find reviewing these guides first to be helpful:
- FAQs for New Contributors: https://make.wordpress.org/core/handbook/tutorials/faq-for-new-contributors/
- Contributing with Code Guide: https://make.wordpress.org/core/handbook/contribute/
- WordPress Coding Standards: https://make.wordpress.org/core/handbook/best-practices/coding-standards/
- Inline Documentation Standards: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/
- Browser Support Policies: https://make.wordpress.org/core/handbook/best-practices/browser-support/
- Proper spelling and grammar related best practices: https://make.wordpress.org/core/handbook/best-practices/spelling/
-->

<!-- Insert a description of your changes here -->

Trac ticket: https://core.trac.wordpress.org/ticket/43095 

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

afercia committed:
Fix aria-haspopup usage in the admin menu.

afercia committed:
Add missing semicolon.

joedolson commented on src/js/_enqueues/admin/common.js:
```diff
@@ -1837,6 +1839,34 @@ $( function() {
+			}
+
+			elements.each( function() {
+				$( this ).removeAttr( 'aria-haspopup' ):
```
Switch from colon to semicolon

afercia replied:
Fixed. My bad vision 😞

afercia committed:
Improve docblock.

afercia committed:
Fix typo.

joedolson reviewed (approved):
Looks good!

joedolson committed:
Remove currentMenuItemHasPopup

With the new aria-haspopup handling, this is no longer relevant. This was required to compensate for menus always having aria-haspopup; now that they have data attributes instead, it's irrelevant.

joedolson committed:
Add handling for aria-expanded states

joedolson commented:
In [r58449]
